### PR TITLE
feat(aws-ecr): add onImageScanCompleted() support

### LINKED ISSUE #4818: aws-ecr to support onImageScanCompleted()
Amazon ECR has [announced](https://aws.amazon.com/about-aws/whats-new/2019/10/announcing-image-scanning-for-amazon-ecr/?nc1=h_ls) the image scanning support and `aws-ecr` should add `onImageScanCompleted()` to trigger the event target from cloudwatch events.

### Use Case

To build a pipeline on image push, scanning on push and notification on scan completed.

More details and demos here: 

https://twitter.com/pahudnet/status/1189556289312935937
and
https://twitter.com/pahudnet/status/1189723775937343488

### Proposed Solution

Add `onImagePushCompleted` in [RepositoryBase](https://github.com/aws/aws-cdk/blob/6e22f262a6266aa6e525f3f43b988f60b1f28481/packages/%40aws-cdk/aws-ecr/lib/repository.ts#L88).

### Other

* [v] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by pahud**
feat(aws-ecr): add onImageScanCompleted() support(fix #4818)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-events/lib/rule.ts [hunk 1]
```diff
@@ -341,15 +341,34 @@ export class Rule extends Resource implements IRule {
       out[key] = value;
     }
 
+    // rename keys in the 'detail' map
+    if (out.detail && Object.keys(out.detail).length > 0) {
+      const origdetail = out.detail;
+      const detailout: any = {};
+      for (let key of Object.keys(origdetail)) {
+        const value = (origdetail as any)[key];
+        if (key === 'repositoryName') {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-events/lib/rule.ts]: There shouldn't be any ECR-specific code in the Events library.

Why do the rename at all, by the way?

File: packages/@aws-cdk/aws-ecr/lib/repository.ts [hunk 2]
```diff
@@ -225,6 +274,19 @@ export interface OnCloudTrailImagePushedOptions extends events.OnEventOptions {
...
 
+/**
+ * Options for the OnImageScanCompleted method
+ */
+export interface OnImageScanCompletedOptions extends events.OnEventOptions {
+  /**
+   * Only watch changes to the image tags spedified.
+   * Leave it undefined to watch the full repository.
+   *
+   * @default undefined
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ecr/lib/repository.ts]: `undefined` is the literal default value, but tell us what the default *behavior* is when the value is left undefined.

File: packages/@aws-cdk/aws-events/lib/rule.ts (hunk 1 above)
Comment (pahud) [reply packages/@aws-cdk/aws-events/lib/rule.ts]: Thanks for the heads up. Have reverted it back.

File: packages/@aws-cdk/aws-ecr/lib/repository.ts (hunk 2 above)
Comment (pahud) [reply packages/@aws-cdk/aws-ecr/lib/repository.ts]: OK. I got it!

**Pull request merged**